shark_turbine/kernel/wave/wave.py

```diff
@@ -0,0 +1,91 @@
+from typing import Callable, Optional
+import inspect
+
+from ..compiler.ir import Context, Operation
+from ..lang import Grid
+from .._support.tracing import (
+    CapturedTrace,
+    CompiledContext,
+    KernelRegionGraph,
+    Launchable,
+)
+from .._support.nodes import *
+
+
+__all__ = ["wave"]
+
+
+def wave():
+    def decorator(f: Callable[[Any], Any]) -> "LaunchableWave":
+        return LaunchableWave(f.__name__, f)
+
+    return decorator
+
+
+class LaunchableWave(Launchable):
+    def __init__(
+        self,
+        name: str,
+        eager_function: Callable[[Any], Any],
+    ):
+        super().__init__(eager_function)
+
+        self.grid_type = Grid[None, None]
+        self._name = name
+        self._f = eager_function
+        self._sig = inspect.signature(eager_function)
+
+    def _trace(self, dump: bool = False) -> CapturedTrace:
+        region_graph = KernelRegionGraph()
+        with CompiledContext(region_graph, grid_type=self.grid_type) as context:
+            custom_ops: dict[str, Type[CustomNode]] = {
+                "construct_register_from_metadata": ConstructRegisterFromMetadataNode,
+                "mma": MmaNode,
+                "read": ReadNode,
+                "write": WriteNode,
+                "reduction": ReductionNode,
+                "placeholder": PlaceholderNode,
+            }
+
+            # Register custom ops
+            for name, op in custom_ops.items():
+                context.register_custom_op(name, op)
+
+            with region_graph.subtracer() as subtracer:
+                root_name, _ = subtracer.trace(self._f)
+                trace = CapturedTrace(region_graph, root_name)
+
+            if dump:
```

Comment:
What do you think of replace dump with a "LLVM_DEBUG" style flag that can be activated based on an incoming argument (like debug-only=)

Reply:
I changed this to use an environment variable `DEBUG`